rust_src/src/fonts.rs

```diff
@@ -97,6 +103,70 @@ impl FontExtraType {
     }
 }
 
+pub type LispFontObjectRef = ExternalPtr<Lisp_Font_Object>;
+
+impl LispFontObjectRef {
+    pub fn as_lisp_obj(self) -> LispObject {
+        LispObject::tag_ptr(self, Lisp_Type::Lisp_Vectorlike)
+    }
+
+    pub fn add_log(self, action: &str, result: LispObject) {
+        let c_str = CString::new(action).unwrap();
```

Comment:
Is this safe?

Reply:
I think so, since the function is only called using hardcoded strings by other functions and not with any user-inputted strings, so unless a function calls it with a broken string there shouldn't be any issues here.

Reply:
Yes, but `intern` is called on it in `font_add_log`. Will the interpreter try to garbage collect that at some point?

Reply:
No allocations made in Rust are safe to the C side. The C code may try to free it.

Reply:
Hm, I hadn't considered that. Is there an established safer way to send strings to c functions?

Reply:
Ok, I just took a few minutes and tracked this to the bottom. The `CString` is not used directly. The contents are `memcpy`ed out. There is no issue here.